### ISSUE TITLE
"Defaulting to user installation" message breaks a lot of stuff

### ISSUE THREAD

**Issue by voxik** [rubygems/rubygems#7082]
It seems that for a lot of commands the "Defaulting to user installation because default installation directory (/usr/share/gems) is not writable." is displayed. E.g.:

~~~
$ gem unpack /builddir/build/SOURCES/json-2.6.3.gem 
Defaulting to user installation because default installation directory (/usr/share/gems) is not writable.
Unpacked gem: '/builddir/json-2.6.3'
~~~

This command has nothing to do with installation directory. The following case is especially harmful and we hit it in Fedora during RPM package buids:

~~~
$ gem spec /builddir/build/SOURCES/json-2.6.3.gem --ruby
Defaulting to user installation because default installation directory (/usr/share/gems) is not writable.
# -*- encoding: utf-8 -*-
# stub: json 2.6.3 ruby lib
# stub: ext/json/ext/generator/extconf.rbext/json/ext/parser/extconf.rbext/json/extconf.rb

Gem::Specification.new do |s|
  s.name = "json".freeze
  s.version = "2.6.3".freeze

  s.required_rubygems_version = Gem::Requirement.new(">= 0".freeze) if s.respond_to? :required_rubygems_version=
  s.metadata = { "bug_tracker_uri" => "https://github.com/flori/json/issues", "changelog_uri" => "https://github.com/flori/json/blob/master/CHANGES.md", "documentation_uri" => "http://flori.github.io/json/doc/index.html", "homepage_uri" => "http://flori.github.io/json/", "source_code_uri" => "https://github.com/flori/json", "wiki_uri" => "https://github.com/flori/json/wiki" } if s.respond_to? :metadata=
  s.require_paths = ["lib".freeze]
  s.authors = ["Florian Frank".freeze]
  s.date = "2022-12-05"
  s.description = "This is a JSON implementation as a Ruby extension in C.".freeze
  s.email = "flori@ping.de".freeze
  s.extensions = ["ext/json/ext/generator/extconf.rb".freeze, "ext/json/ext/parser/extconf.rb".freeze, "ext/json/extconf.rb".freeze]
  s.extra_rdoc_files = ["README.md".freeze]
  s.files = ["CHANGES.md".freeze, "LICENSE".freeze, "README.md".freeze, "VERSION".freeze, "ext/json/ext/fbuffer/fbuffer.h".freeze, "ext/json/ext/generator/depend".freeze, "ext/json/ext/generator/extconf.rb".freeze, "ext/json/ext/generator/generator.c".freeze, "ext/json/ext/generator/generator.h".freeze, "ext/json/ext/parser/depend".freeze, "ext/json/ext/parser/extconf.rb".freeze, "ext/json/ext/parser/parser.c".freeze, "ext/json/ext/parser/parser.h".freeze, "ext/json/ext/parser/parser.rl".freeze, "ext/json/extconf.rb".freeze, "json.gemspec".freeze, "lib/json.rb".freeze, "lib/json/add/bigdecimal.rb".freeze, "lib/json/add/complex.rb".freeze, "lib/json/add/core.rb".freeze, "lib/json/add/date.rb".freeze, "lib/json/add/date_time.rb".freeze, "lib/json/add/exception.rb".freeze, "lib/json/add/ostruct.rb".freeze, "lib/json/add/range.rb".freeze, "lib/json/add/rational.rb".freeze, "lib/json/add/regexp.rb".freeze, "lib/json/add/set.rb".freeze, "lib/json/add/struct.rb".freeze, "lib/json/add/symbol.rb".freeze, "lib/json/add/time.rb".freeze, "lib/json/common.rb".freeze, "lib/json/ext.rb".freeze, "lib/json/generic_object.rb".freeze, "lib/json/pure.rb".freeze, "lib/json/pure/generator.rb".freeze, "lib/json/pure/parser.rb".freeze, "lib/json/version.rb".freeze]
  s.homepage = "http://flori.github.com/json".freeze
  s.licenses = ["Ruby".freeze]
  s.rdoc_options = ["--title".freeze, "JSON implementation for Ruby".freeze, "--main".freeze, "README.md".freeze]
  s.required_ruby_version = Gem::Requirement.new(">= 2.3".freeze)
  s.rubygems_version = "3.5.0.dev".freeze
  s.summary = "JSON Implementation for Ruby".freeze
end
~~~

Please notice that the output is polluted by the message.

@duckinator PTAL. Thx

**Comment by voxik** [comment]
Please note that this happens during the RPM package builds. This is bare minimum environment. So the home directory is essentially empty.

**Comment by voxik** [comment]
This is related to #5327

**Comment by duckinator** [comment]
Thanks for bringing this to my attention @voxik. It would ideally only print that message on `gem install` and `gem update`, I think.

**Comment by duckinator** [comment]
@voxik I just merged #7207. I'll ask for that change to be included in the next release. Thanks for making sure I was aware of this!
